### PR TITLE
Controllers : Wiring the Endpoint to actions and actually replace the legacy system

### PR DESCRIPTION
**Pull request by samcake**
- Introduce the ActionEndpoint to add the value to the current action State of the UserINputMapper

- have a spearate list of "externalActionStates" in the userINputMApper to collect the values mordifed from outside (the ScriptingController) and combine thme with the legacy system correctly

- put _standardDevice as a regular _proxyDevice so not having to differenciate the code between standard and non standard

- typos

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: f0034844e7b5ba938b7234239f2f2683f99fd917 available at: https://gist.github.com/13325c0b30b19320df79<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6098.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6098.dmg

File: libraries/controllers/src/controllers/ScriptingInterface.cpp [hunk 1]
```diff
@@ -94,6 +94,25 @@ namespace controller {
         Endpoint::Pointer _second;
     };
 
+    class ActionEndpoint : public Endpoint {
+    public:
+        ActionEndpoint(const UserInputMapper::Input& id = UserInputMapper::Input(-1))
```
Comment (jherico) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: should default to INVALID_INPUT

File: libraries/controllers/src/controllers/ScriptingInterface.cpp [hunk 2]
```diff
@@ -139,7 +158,8 @@ namespace controller {
 
             // Create the endpoints
             // FIXME action endpoints need to accumulate values, and have them cleared at each frame
```
Comment (jherico) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: this is fixed, no?

File: libraries/controllers/src/controllers/ScriptingInterface.cpp [hunk 3]
```diff
@@ -139,7 +158,8 @@ namespace controller {
...
-            _endpoints[actionInput] = std::make_shared<VirtualEndpoint>();
+           // _endpoints[actionInput] = std::make_shared<VirtualEndpoint>();
```
Comment (jherico) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: dead code

File: libraries/controllers/src/controllers/UserInputMapper.cpp [hunk 4]
```diff
@@ -359,15 +364,15 @@ void UserInputMapper::assignDefaulActionScales() {
...
-    _actionStates[SHIFT] = 1.0f; // on
-    _actionStates[ACTION1] = 1.0f; // default
-    _actionStates[ACTION2] = 1.0f; // default
-    _actionStates[TRANSLATE_X] = 1.0f; // default
-    _actionStates[TRANSLATE_Y] = 1.0f; // default
-    _actionStates[TRANSLATE_Z] = 1.0f; // default
-    _actionStates[ROLL] = 1.0f; // default
-    _actionStates[PITCH] = 1.0f; // default
-    _actionStates[YAW] = 1.0f; // default
+    _actionScales[SHIFT] = 1.0f; // on
```
Comment (jherico) [inline libraries/controllers/src/controllers/UserInputMapper.cpp]: whoops.  sorry

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0fc04ab2972083df9daabe4fae6b14b0c01c2464 available at: https://gist.github.com/13325c0b30b19320df79<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6098.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6098.dmg

**Pull request merged**